File: backend/src/main/java/bpr/fitness/companion/backend/model/dto/WeekRoutine.java
package bpr.fitness.companion.backend.model.dto;


import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import java.util.Set;

@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
public class WeekRoutine {
    private Long id;
    private String name;
    private String notes;
    private Set<Long> dayRoutines;
    private Long userId;
}

File: FrontEnd/app/src/main/java/com/example/fitnesscompanion_front/model/WeekRoutine.kt
package com.example.fitnesscompanion_front.model

data class WeekRoutine(
    val id: Int = 1,
    val name: String = "",
    val notes: String = "",
    val dayRoutines: List<DayRoutine> = emptyList(),
    val userId: Int = 1
)

data class WeekRoutineRequest(
    val name: String,
    val notes: String,
    val userId: Int
)